spec/expression.dd

```diff
@@ -21,13 +21,44 @@ full expression of $(I expr$(SUBSCRIPT 1)). Otherwise, $(I expr) is its own full
 
 $(P Each expression has a unique full expression.)
 
-Example: in the statement `return f() + g() * 2;`, the full expression of `g() * 2` is `f() + g() * 2`, but not the
-full expression of `f() + g()` because the latter is not parsed as a subexpression.
+$(P Example: in the statement `return f() + g() * 2;`, the full expression of `g() * 2` is `f() + g() * 2`, but not the
+full expression of `f() + g()` because the latter is not parsed as a subexpression.)
 
-Note: Although the definition is straightforward, a few subtleties exist related to function literals. In the
+$(P Note: Although the definition is straightforward, a few subtleties exist related to function literals. In the
 statement `return (() => x + f())() * g();`, the full expression of `f()` is `x + f()`, not the expression passed
-to `return`. This is because the parent of `x + f()` has function literal type, not expression type.
-
+to `return`. This is because the parent of `x + f()` has function literal type, not expression type.)
+
+$(DDOC_ANCHOR define-lvalue)$(P $(B Definition) ($(DOUBLEQUOTE Lvalue)): The following expressions, and no
+others, are called lvalue expressions or lvalues:)
+$(OL
+$(LI `this` inside `struct` and `union` member functions;)
+$(LI the result of the $(I DotIdentifier) grammatical construct `.` (left side may be missing) when the
+rightmost side of the dot is the name of a field (whether direct or `static`), variable, or function;)
+$(LI the result of the following expressions:
+$(UL
+$(LI built-in unary operators `+` (when applied to an lvalue), `*`, `++` (prefix only), `--` (prefix only);)
+$(LI built-in indexing operator `[]` (but not the slicing operator);)
+$(LI built-in assignment binary operators, i.e. `=`, `+=`, `*=`, `/=`, `%=`, `&=`, `|=`, `^=`, `~=`,
+`<<=`, `>>=`, `>>>=`, and `^^=`;)
+$(LI the ternary operator $(I e) `?` $(I e$(SUBSCRIPT 1)) `:` $(I e$(SUBSCRIPT 2)) under the following
+circumstances:)
+$(OL 
+    $(LI $(I e$(SUBSCRIPT 1)) and $(I e$(SUBSCRIPT 1)) are lvalues of the same type; OR)
+    $(LI One of $(I e$(SUBSCRIPT 1)) and $(I e$(SUBSCRIPT 2)) is an lvalue of type `T` and the other has
+    and `alias this` converting it to `ref T`.))
+$(LI `mixin` expressions if and only if the compilation of the expression resulting from compiling
+the argument(s) to `mixin` is an lvalue;)
+$(LI `cast(U)` expressions applied to lvalues of type `T` when `T*` is implicitly convertible to `U*`;)
+$(LI `cast()` and `cast(`$(I qualifier list)`)` when applied to an lvalue.)
+)))
```

Comment:
Missing:
- Identifiers referencing local variables.
- Results of `ref` function calls.

Reply:
Thanks! Those got removed during some text reshuffling.